### DISCUSSION TITLE
How to choose settings file?

### DISCUSSION THREAD

**Discussion by qarmin** [astral-sh/ruff#3399] [Q&A] [answered]
Hi,
In my repository I have this structure:
```
- /
  - app1/
     - *.py
  - app2/
     - *.py
  - ruff.toml
``` 
In local machine, when I run ruff from `app1`, `app2`, or `/` folder, then settings file is properly read, but when I do the same in CI, then this not works.

I don't have any idea what is the issue, so I tried to manually set setting file, but I didn't found any option to do this like `ruff --settings-file ../ruff.toml` - it is possible to do this?

**Comment by charliermarsh** [comment, marked as answer]
Yeah, you can pass the file directly via `ruff check --config /path/to/ruff.toml`, which takes the path to the configuration file. I would expect what you have there to work, though, as long as everything is being copied over including the configuration file.
